llvm/test/lab2/tatiana_bend/Mrrrrr.ll

```diff
@@ -0,0 +1,110 @@
+; RUN: opt -load-pass-plugin=%llvmshlibdir/MrrrBend%shlibext -passes=MrrrBend -S %s | FileCheck %s
+
+; void foo() {
+;     return;
+; }
+
+; int bar(int z) {
+;     return z + z;
+; }
+
+; int baz(int k) {
+;     if (k % 2) {
+;         return k / 2;
+;     } else {
+;         return (k + 1) / 2;
+;     }
+; }
+
+; int Mrrr(int x){
+;     instrument_start();
+;     x=x+5;
+;     x=x+10;
+;     instrument_end();
+;     return x;
+; }
+
+; CHECK-LABEL: @foo
+; CHECK: call void @instrument_start()
+; CHECK-NEXT: call void @instrument_end()
+; CHECK-NEXT: ret void
+
+define dso_local void @foo() {
+entry:
+  ret void
+}
+
+; CHECK-LABEL: @bar
+; CHECK: call void @instrument_start()
+; CHECK-NEXT: %z.addr = alloca i32, align 4
+; CHECK: call void @instrument_end()
+; CHECK-NEXT: ret i32 %add
+
+define dso_local i32 @bar(i32 noundef %z) {
+entry:
+  %z.addr = alloca i32, align 4
```

Comment:
Please insert instrument_start at the beginning but after static data allocations.